src/main/java/com/arpnetworking/clusteraggregator/aggregation/StreamingAggregator.java

```diff
@@ -217,72 +197,62 @@ public void preRestart(final Throwable reason, final Optional<Object> message) t
 
     private void processAggregationMessage(final Messages.StatisticSetRecord data) {
         final CombinedMetricData metricData = CombinedMetricData.Builder.fromStatisticSetRecord(data).build();
-        //First message sets the data we know about this actor
-        if (!_initialized) {
-            _period = metricData.getPeriod();
-            _cluster = metricData.getCluster();
-            _metric = metricData.getMetricName();
-            _service = metricData.getService();
-            _dimensions = dimensionsToMap(data);
-            _resultBuilder = new AggregatedData.Builder()
-                    .setHost(createHost())
-                    .setPeriod(_period)
-                    .setPopulationSize(1L)
-                    .setSamples(Collections.emptyList())
-                    .setStart(DateTime.now().hourOfDay().roundFloorCopy())
-                    .setValue(new Quantity.Builder().setValue(0d).build());
 
+        // First message sets the data we know about this actor
+        initialize(data, metricData);
 
-            _initialized = true;
-            LOGGER.debug()
-                    .setMessage("Initialized aggregator")
-                    .addContext("actor", self())
-                    .log();
-        } else if (!(_period.equals(metricData.getPeriod())
-                && _cluster.equals(metricData.getCluster())
-                && _service.equals(metricData.getService())
-                && _metric.equals(metricData.getMetricName()))) {
-            LOGGER.error()
-                    .setMessage("Received a work item for another aggregator")
-                    .addData("workItem", data)
-                    .addContext("actor", self())
-                    .log();
-        }
-        //Find the time bucket to dump this in
+        // Find the time bucket to dump this in
         final DateTime periodStart = DateTime.parse(data.getPeriodStart());
         if (_aggBuckets.size() > 0 && _aggBuckets.getFirst().getPeriodStart().isAfter(periodStart)) {
-            //We got a bit of data that is too old for us to aggregate.
+            // We got a bit of data that is too old for us to aggregate.
             LOGGER.warn()
```

Comment:
We should probably rate limit this.  In the past I've seen it saturate logs.

Reply:
Fixed